pdns/ixfrdist.cc

```diff
@@ -316,6 +316,9 @@ void updateThread(const string& workdir, const uint16_t& keep, const uint16_t& a
               soa = getRR<SOARecordContent>(dr);
             }
           }
+          if (axfrMaxRecords != 0 && nrecords > axfrMaxRecords) {
```

Comment:
Did you ponder moving this into `AXFRRetriever`, so we can reuse it later for `RPZ` or other regular `AXFR` stuff?

Reply:
Yes, let's move this to a new ticket :)